doc/guides/maintaining-the-build-files.md

```diff
@@ -0,0 +1,199 @@
+# Maintaining the Build files
+
+This document explains how to maintain the build files in the codebase.
+
+## Overview
+
+On how to build the Node.js core, see [Building Node.js](../../BUILDING.md).
+
+There are three main build files that will be directly run to build Node.js:
+
+On Windows:
+
+- `vcbuild.bat`: A Windows Batch Script that locates build tools, provides
+  a subset of the targets avilable in the [Makefile](#makefile), and a few
+  targets of its own. This script also calls [configure](#configure) before
+  building the Node.js binary. For a detailed guide on this script,
+  see [vcbuild.bat](#vcbuild.bat).
+
+On unix-like systems:
+
+- `configure`: A Python 2 script that detects system capabilities and runs
+  [GYP][]. It generates `config.gypi` which includes parameters used
+  by GYP to create platform-dependent build files. Its output is usually
+  in one of these formats: Makefile, MSbuild, ninja, or XCode project files.
```

Comment:
When you say `Its output`, do you mean `configure`'s output or GYP's output? If GYP, maybe move it to its own section, as GYP is used on Windows as well.